src/ucp/wireup/select.c

```diff
@@ -1507,18 +1509,39 @@ ucp_wireup_add_fast_lanes(ucp_worker_h worker,
     ucp_context_h context      = worker->context;
     const double max_ratio     = 1. / context->config.ext.multi_lane_max_ratio;
     ucs_status_t status;
+    ucp_md_index_t md_index;
     double lane_bw;
     const ucp_wireup_select_info_t *sinfo;
+    const ucp_wireup_lane_desc_t *lane_desc;
+    const uct_md_attr_v2_t *md_attr;
+
+    /* Calculate max BW of existing lanes of the corresponding mem type */
+    for (lane_desc = select_ctx->lane_descs;
+     lane_desc < select_ctx->lane_descs + select_ctx->num_lanes; ++lane_desc) {
+        if (lane_desc->rsc_index != UCP_NULL_RESOURCE) {
+            md_index = context->tl_rscs[lane_desc->rsc_index].md_index;
+            md_attr  = &context->tl_mds[md_index].attr;
+            if (md_attr->reg_mem_types & lane_reg_mem_types) {
+                lane_bw = ucp_wireup_get_lane_bw(worker, lane_desc->rsc_index,
+                                                 lane_desc->addr_index,
+                                                 select_params->address);
+                max_bw  = ucs_max(lane_bw, max_bw);
+            }
+        }
+    }
```

Comment:
how about keeping max_bw in ucp_wireup_select_context_t?
in any case, seems redundant to do bot this loop and also the loop in lines 1534-1539

Reply:
I tried to do that, but the problem is that we're dealing with a mem_type bitmap, not a single mem_type. The `ucp_wireup_criteria_t` in `bw_info` has a `reg_mem_types` bitmap. Though, it seems like that bitmap is set to only one mem_type in the existing code.